### ISSUE TITLE
Sending multiple video stream to one user

### ISSUE THREAD

**Issue by nodminger** [sipsorcery-org/sipsorcery#1031]
I am working on a multiple user streaming application. Using multiple Peer Connections, I am able to forward the video stream between peers using this code.
```
PC.OnRtpPacketReceived += (IPEndPoint rep, SDPMediaTypesEnum media, RTPPacket rtpPkt) =>
{
    foreach (var pcs in session[id])
    {
      pcs.SendRtpRaw(rtpPkt.Payload, rtpPkt.Header.Timestamp, rtpPkt.Header.MarkerBit, rtpPkt.Header.PayloadType);
    }
};
```
_Basically in this code, up on receiving video from one user I am forwarding it to other Peer Connections which I have already saved in a list_

But when using this code multiple streams are combined and I am facing issue when playing it on the user end(JavaScript - web app). When there are only 2 users, the app works fine. Because it sends only one videos between both.

A -> B
B -> A

But when there are multiple users, it has to go like this

A - > B,C
B -> A,C
C -> A,B

Since all the packets are sent as RTP packets directly to the user, I think it getting merged and facing some issue. Is there a way to split the track and send multiple users video stream to one user?

In this link [https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/addTrack](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/addTrack) they gave an example(JavaScript) like this `addTrack(track, stream1, stream2, /* …, */ streamN)` for adding multiple stream on a single track. Is is possible to do the same for forwarding RTP packets in Sipsorcery?

**Comment by sipsorcery** [comment]
Yes, it can be done but only in a very crude fashion. You can create completely separate RTCPeerConnections with each peer and send the same video stream tothem all. Likewise you'll receive individual streams from each of the peers. It's not too bad with 3 peers but is going to become very bandwith and CPU heavy very quickly as you add peers.

The typical solution for this is to use a Selective Forwarding Unit (SFU) server. It's essentially a server that receives all the stream from each peer and uses some logic to only forward the streams to each peer as required. Another option is to use a media mixer such that all the peers forward to the server and it combines them all into a single audio and video stream. A mixer can require a powerful server so is unlikely to be free.
